src/key.rs

```diff
@@ -402,7 +427,32 @@ impl From<ffi::PublicKey> for PublicKey {
     }
 }
 
-serde_impl_from_slice!(PublicKey);
+#[cfg(feature = "serde")]
+impl ::serde::Serialize for PublicKey {
+    fn serialize<S: ::serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        if s.is_human_readable() {
+            s.collect_str(self)
+        } else {
+            s.serialize_bytes(&self.serialize())
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> ::serde::Deserialize<'de> for PublicKey {
+    fn deserialize<D: ::serde::Deserializer<'de>>(d: D) -> Result<PublicKey, D::Error> {
+        if d.is_human_readable() {
+            d.deserialize_str(super::serde_util::FromStrVisitor::new(
+                "an ASCII hex string representing a public key"
+            ))
+        } else {
+            d.deserialize_bytes(super::serde_util::BytesVisitor::new(
+                "a bytestring representing a public key",
+                PublicKey::from_slice
```

Comment:
Isn't this a breaking change? currently bytes will deserialize as hex: https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/macros.rs#L116-L125

(I'm not saying it's bad, just trying to know if this might break downstream users)

Reply:
I don't think it is!

The lines you commented on are within the _else_ branch of `d.is_human_readable` whereas the code you linked to is within `d.is_human_readable`.

Also, I think the linked piece of code was never actually called because I deleted a legacy of it in c2fd5cefa4714effe51ae052faa4ef132aa526ec and all tests still pass!

Reply:
hmm I see, you're right.